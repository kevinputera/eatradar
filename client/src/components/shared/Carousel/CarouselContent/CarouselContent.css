.carousel-content {
  flex-grow: 1;
}

.carousel-content-link {
  margin-left: 10px;
}

.carousel-content-author {
  text-align: end;
  font-style: italic;
}
