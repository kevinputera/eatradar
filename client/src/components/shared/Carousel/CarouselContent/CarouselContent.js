import React from 'react';
import { Divider } from '@blueprintjs/core';

import './CarouselContent.css';

function CarouselContent(props) {
  const content = props.contents[props.index];
  return (
    <div className="carousel-content">
      {content.title && (
        <>
          <div className="carousel-content-title">{content.title}</div>
          <Divider />
        </>
      )}
      <div className="carousel-content-post">
        {content.post}
        {content.link && (
          <a
            className="carousel-content-link"
            href={content.link}
            target="_blank"
            rel="noopener noreferrer"
          >
            Read more
          </a>
        )}
      </div>
      {content.author && (
        <div className="carousel-content-author">{content.author}</div>
      )}
    </div>
  );
}

export default CarouselContent;
