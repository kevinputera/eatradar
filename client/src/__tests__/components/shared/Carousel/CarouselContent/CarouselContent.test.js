import React from 'react';
import { create } from 'react-test-renderer';

import CarouselContent from '../../../../../components/shared/Carousel/CarouselContent/CarouselContent';

describe('tests for CarouselContent component', () => {
  test('render must match snapshot', () => {
    const post = 'A post';
    const title = 'A title';
    const link = 'https://www.eatradar.com';
    const author = 'An author';

    const tree = create(<CarouselContent content={{ post }} />).toJSON();
    expect(tree).toMatchSnapshot();

    const treeWithTitle = create(
      <CarouselContent content={{ post, title }} />
    ).toJSON();
    expect(treeWithTitle).toMatchSnapshot();

    const treeWithLink = create(
      <CarouselContent content={{ post, link }} />
    ).toJSON();
    expect(treeWithLink).toMatchSnapshot();

    const treeWithAuthor = create(
      <CarouselContent content={{ post, author }} />
    ).toJSON();
    expect(treeWithAuthor).toMatchSnapshot();
  });
});
