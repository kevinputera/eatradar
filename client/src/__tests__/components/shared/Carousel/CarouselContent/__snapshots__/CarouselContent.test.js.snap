// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for CarouselContent component render must match snapshot 1`] = `
<div
  className="carousel-content"
>
  <div
    className="carousel-content-post"
  >
    A post
  </div>
</div>
`;

exports[`tests for CarouselContent component render must match snapshot 2`] = `
<div
  className="carousel-content"
>
  <div
    className="carousel-content-title"
  >
    A title
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="carousel-content-post"
  >
    A post
  </div>
</div>
`;

exports[`tests for CarouselContent component render must match snapshot 3`] = `
<div
  className="carousel-content"
>
  <div
    className="carousel-content-post"
  >
    A post
    <a
      className="carousel-content-link"
      href="https://www.eatradar.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Read more
    </a>
  </div>
</div>
`;

exports[`tests for CarouselContent component render must match snapshot 4`] = `
<div
  className="carousel-content"
>
  <div
    className="carousel-content-post"
  >
    A post
  </div>
  <div
    className="carousel-content-author"
  >
    An author
  </div>
</div>
`;
